pkg/client/restclient/request.go

```diff
@@ -551,13 +553,32 @@ func (r *Request) Body(obj interface{}) *Request {
 		if reflect.ValueOf(t).IsNil() {
 			return r
 		}
-		data, err := runtime.Encode(r.serializers.Encoder, t)
-		if err != nil {
-			r.err = err
-			return r
+		switch typedObj := t.(type) {
+		// hack for eviction
+		// TODO: remove this when we find a better way to encode eviction properly.
+		case *policy.Eviction:
+			codec, err := testapi.GetCodecForObject(typedObj)
```

Comment:
You can't use testapi in production code. It's doesn't negotiate the serializer with the API server.